lib/core/network/repository/proposal_repository.dart

```diff
@@ -10,35 +11,72 @@ class ProposalRepository {
+
+    final List<ProposalModel> allProposals = [];
+
+    for (final Result<Map<String, dynamic>, HyphaError> result in futureResults) {
+
+      if (result.isValue) {
+        final Map<String, dynamic> response = result.asValue!.value;
 
-      try {
-        final Map<String, dynamic> response = result.valueOrCrash;
-        final List<dynamic> proposalsData = response['data']['queryDao'];
-        final List<ProposalModel> proposals = proposalsData.expand((dao) {
-          final String daoName = dao['details_daoName_n'];
-          final List<dynamic> proposals = dao['proposal'] as List<dynamic>;
-
-          return proposals.map((dynamic proposal) {
-            return ProposalModel.fromJson({...{'daoName': daoName}, ...proposal});
-          });
-        }).toList();
-
-        return Result.value(proposals);
-      } catch (e, stackTrace) {
-        LogHelper.e('Error parsing data into proposal model', error: e, stacktrace: stackTrace);
-        return Result.error(HyphaError.generic('Error parsing data into proposal model'));
+        if (response['errors'] != null) {
+          LogHelper.e('GraphQL query failed', error: response['errors']);
+          return Result.error(HyphaError.api('GraphQL query failed'));
+        }
+
+        try {
+          final List<ProposalModel> proposals = _parseProposalsFromResponse(response);
+          allProposals.addAll(proposals);
+        } catch (e, stackTrace) {
+          LogHelper.e('Error parsing data into proposal model', error: e, stacktrace: stackTrace);
+          return Result.error(HyphaError.generic('Error parsing data into proposal model'));
+        }
+      } else {
+        LogHelper.e('GraphQL query failed', error: result.asError!.error);
+        return Result.error(result.asError!.error);
       }
-    } else {
-      LogHelper.e('GraphQL query failed', error: result.asError!.error);
-      return Result.error(result.asError!.error);
     }
+
+    sortProposals(allProposals);
+    return Result.value(allProposals);
+  }
+
+  List<ProposalModel> _parseProposalsFromResponse(Map<String, dynamic> response) {
+    final List<dynamic> proposalsData = response['data']['queryDao'];
+    return proposalsData.expand((dao) {
+      final String daoName = dao['details_daoName_n'];
+      final List<dynamic> proposals = dao['proposal'] as List<dynamic>;
+      return proposals.map((dynamic proposal) {
+        return ProposalModel.fromJson({...{'daoName': daoName}, ...proposal});
+      });
+    }).toList();
+  }
+
+  void sortProposals(List<ProposalModel> proposals) {
+    proposals.sort((a, b) {
+      final daoNameComparison = a.daoName.compareTo(b.daoName);
+      if (daoNameComparison != 0) {
+        return daoNameComparison;
+      }
+
+      final now = DateTime.now();
+
+      final isAExpired = a.expiration != null && a.expiration!.isBefore(now);
+      final isBExpired = b.expiration != null && b.expiration!.isBefore(now);
+
```

Comment:
if we have the concept that a proposal can be expired then that should be handled in a utility function so we don't repeat this code everywhere (isBefore(now))

Reply:
I recommend an extension on ProposalModel "isExpired"